catalog/core/catalog-core-versioning/versioning-common/src/main/java/ddf/catalog/core/versioning/impl/MetacardVersionImpl.java

```diff
@@ -183,44 +183,77 @@ public MetacardVersionImpl(String id, Metacard sourceMetacard, Action action, Su
         this.setTags(Collections.singleton(VERSION_TAG));
     }
 
+    //@formatter:off
     private byte[] getVersionType(Metacard sourceMetacard) {
-        try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
-                ObjectOutputStream oos = new ObjectOutputStream(baos)) {
-            oos.writeObject(sourceMetacard.getMetacardType());
-            return baos.toByteArray();
-        } catch (IOException e) {
+        try {
+            return METACARD_TYPE_CACHE.get(sourceMetacard.getMetacardType()
+                    .hashCode(), () -> {
+                    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                        ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+                        oos.writeObject(sourceMetacard.getMetacardType());
+                        METACARD_TYPE_CACHE.put(sourceMetacard.getMetacardType()
+                                .hashCode(), baos.toByteArray());
+                        return baos.toByteArray();
+                    }
+                });
+        } catch (ExecutionException e) {
+            if (LOGGER.isDebugEnabled()) {
```

Comment:
Printing the source metacard. Its string representation could possibly be heavy if an implementation prints all attribute and type information.